src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -553,7 +553,17 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
                     resultRight = CreateConversion(right, best.RightConversion, signature.RightType, diagnostics);
                     resultConstant = FoldBinaryOperator(node, resultOperatorKind, resultLeft, resultRight, resultType.SpecialType, diagnostics, ref compoundStringLength);
                     HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                    hasErrors = isObjectEquality && !BuiltInOperators.IsValidObjectEquality(Conversions, leftType, leftNull, rightType, rightNull, ref useSiteDiagnostics);
+
+                    bool leftDefault = left.IsLiteralDefault();
+                    bool rightDefault = right.IsLiteralDefault();
+
+                    if (isObjectEquality && (leftDefault || rightDefault))
+                    {
+                        CheckFeatureAvailability(node, MessageID.IDS_DefaultEquality, ref useSiteDiagnostics);
```

Comment:
> CheckFeatureAvailability(node, MessageID.IDS_DefaultEquality, ref useSiteDiagnostics); [](start = 23, length = 87)

I think this is a abuse of this helper, we probably should move it to conversions, the only existing consumer at the moment.

Reply:
To clarify, I don't think useSiteDiagnostics should be used for this purpose

---
In reply to: [170335206](https://github.com/dotnet/roslyn/pull/24913#discussion_r170335206) [](ancestors = 170335206)

Reply:
We should also file a bug for span implementation and get rid of the helper completely.

---
In reply to: [170335434](https://github.com/dotnet/roslyn/pull/24913#discussion_r170335434) [](ancestors = 170335434,170335206)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/25028](https://github.com/dotnet/roslyn/issues/25028)

---
In reply to: [170336116](https://github.com/dotnet/roslyn/pull/24913#discussion_r170336116) [](ancestors = 170336116,170335434,170335206)